parsec/interfaces/dtd/insert_function.c

```diff
@@ -390,17 +390,14 @@ parsec_dtd_taskpool_destructor(parsec_dtd_taskpool_t *tp)
 
     /* Destroy the data repositories for this object */
     for (i = 0; i < PARSEC_DTD_NB_TASK_CLASSES; i++) {
-        const parsec_task_class_t *tc = tp->super.task_classes_array[i];
+        parsec_task_class_t *tc = (parsec_task_class_t *) tp->super.task_classes_array[i];
         /* Have we reached the end of known functions for this taskpool? */
         if( NULL == tc ) {
             continue;
         }
-        parsec_dtd_task_class_t   *dtd_tc = (parsec_dtd_task_class_t *)tc;
 
-        uint64_t fkey = (uint64_t)(uintptr_t)dtd_tc->cpu_func_ptr + tc->nb_flows;
-        parsec_dtd_release_task_class( tp, fkey );
+        parsec_dtd_task_class_release( (parsec_taskpool_t*)tp, tc );
```

Comment:
With either Chameleon or Dplasma, i gives no new memory leak from what i can tell. I removed it because it actually created double-free issues, and from what I gather, what template_release does is covered by task_class_release.